features/270-next-skycli/api.md

```diff
@@ -0,0 +1,139 @@
+## Feature Overview
+
+In next, we will have cli tools for user to manage the cluster's applications.
+Since we will only have skycli in the first version, skycli should be able to
+do almost all the operations of v1 portal.
+
+The following spec will cover the EE version first.
+
+
+## User scenarios
+
+- Cluster admin can create cluster user
+- Cluster admin assign user to app
+- Cluster user can create app
+- Cluster user can view, update and delete their apps config
+- Cluster user deploy CF to their apps
+- Cluster user can update apps custom domain setting, add new custom domain to app
+- Cluster admin can do everything in the cluster
+
+## Commands naming guidelines
+
+**Format**
+
+```
+skycli [group] [verb/action] [parameters]
+```
+
+**Some common operations**
+
+| Operation | Verb        |
+|-----------|-------------|
+| Create    | create      |
+| Read      | view / list |
+| Update    | update      |
+| Delete    | remove      |
+
+
+## Commands
+
+- skycli auth
+    - [skycli auth login](commands.md#skycli-auth-login)
+    - [skycli auth logout](commands.md#skycli-auth-logout)
+
+- skycli config
+    - [skycli config view](commands.md#skycli-config-view)
+    - [skycli config set-app [APP_NAME]](commands.md#skycli-config-set-app)
```

Comment:
If account and cluster information are in user's directory, and per app information are in per app's directory, isn't "change app" will be done by implicit action of moving into an app directory, and there should be no way for users to "change an app" without moving into the app's directory?